### PR TITLE
chore: use @carbon/styles (v11)

### LINKED ISSUE #1253: When will Carbon v11 be available for Svelte?
(title).
Thanks.

### PR DESCRIPTION
**Pull request by gregorw**
Reworking #1496. See https://carbon-components-svelte-v11.onrender.com for actual results.

This PR introduces v11 Carbon styles. As a first step, I would make sure that all existing components still work, before introducing new components or changing the technology stack in any other ways.

- [x] @carbon/styles
- [x] @ibm/plex

Here’s the list of components that need to be verified:

- [x] Accordion
- [x] AspectRatio
- [x] Breadcrumb
- [x] Breakpoint
- [x] Button
- [x] ButtonSet
- [x] Checkbox
- [x] ClickableTile
  - [ ] Light variant is not working yet. Uses layer tokens.
- [ ] CodeSnippet
  - [ ] Copy button is misaligned
- [x] ComboBox
- [x] ComposedModal
- [x] ContentSwitcher
- [ ] ContextMenu
- [x] CopyButton
  - [ ] Tooltip is misaligned
- [x] DataTable
  - [ ] `stickyHeader` has a bug in @carbon/styles most probably,  `rem(300px)` is not recognized by browser.
- [x] DatePicker
- [x] Dropdown
- [x] ExpandableTile
- [x] FileUploader
- [x] FluidForm
- [x] Form
  - [ ] Docs need `Stack` component for vertical spacing
- [x] Grid
  - [ ] Needs renaming to `FlexGrid` at some point to clear `Grid` for css grids.
- [x] ImageLoader
- [x] InlineLoading
- [x] InlineNotification
  - [ ] Docs need `Stack` component for vertical spacing
- [x] Link
- [x] Loading
- [x] LocalStorage
  - [ ] Dark mode toggle needs fixing. See `Theme`/`GlobalTheme`
- [x] Modal
- [x] MultiSelect
- [x] NumberInput
- [x] OrderedList
- [x] OverflowMenu
- [x] Pagination
- [x] PaginationNav
- [x] PasswordInput
- [x] Popover
- [x] ProgressBar
- [x] ProgressIndicator
- [x] RadioButton
- [x] RadioTile
- [x] RecursiveList
- [x] Search
- [x] Select
- [x] SelectableTile
- [x] SkeletonPlaceholder
- [x] SkeletonText
- [x] Slider
- [x] StructuredList
  - [ ] Selectable version needs double-checking
- [x] Tabs
- [x] Tag
  - [ ] Interactive needs fixing
- [x] TextArea
- [x] TextInput
- [ ] Theme
- [x] Tile
- [x] TimePicker
- [x] ToastNotification
  - [ ] Docs need `Stack` component for vertical spacing
- [x] Toggle
- [x] Tooltip
- [x] TooltipDefinition
- [ ] TooltipIcon
- [x] TreeView
- [x] ~~Truncate~~ _removed_
- [x] UIShell
  - [ ] HeaderSearch needs to be double-checked
- [x] UnorderedList

Fixes #1253.

—

Other resources and considerations

- [ ] Components
  - [ ] [`Stack`](https://react.carbondesignsystem.com/?path=/story/components-form--default)
  - [ ] [Notifications](https://carbondesignsystem.com/migrating/guide/design#notifications-breaking)
    - [ ] [`ActionableNotification`](https://carbondesignsystem.com/components/notification/usage/#actionable-notifications)
    - [ ] Remove action from `InlineNotification`
  - [x] [`Popover`](https://carbondesignsystem.com/migrating/guide/design#popover-new)
  - [x] [`Tooltip`](https://carbondesignsystem.com/migrating/guide/design#tooltip-breaking): Refactor to use Popover
  - [ ] [`Toggletip`](https://carbondesignsystem.com/migrating/guide/design#toggletip-new)
  - [x] Refactor [Tabs](https://carbondesignsystem.com/migrating/guide/design#tabs-breaking)
  - [ ] [`UIShell`](https://carbondesignsystem.com/migrating/guide/design#ui-shell-updated)
- [x] Sizing
- [ ] Type tokens
- [ ] Color tokens
- [ ] Theming
  - [ ] `Theme` component (see also fac78ee4aad272e60b6ebfbdc64fdc2d7ef95471)
  - [ ] `Layer` component
  - [ ] Remove `light` variants

### CODE REVIEW

**Comment by theetrain** [comment]
Thanks again for taking the time; I see you even refactored Tabs to replace the dropdown styles with a scrolling tabs list, which matches v11 implementation details.

To ensure I understand, is the proposal here to:

1. First prove that components work with v11 styles, and refactor them accordingly
2. Then, figure out an improved way for consumers to set up Carbon CSS in their projects
3. Ship these components, then continue work towards `carbon-components-svelte@v1`
4. Afterwards (out of scope for now), we set up `svelte-package`, types, and other v1 goals

If so, I'm on board with this.

**Comment by gregorw** [comment]
> 1. First prove that components work with v11 styles, and refactor them accordingly

Exactly, this should be our first priority. This change should then be merged to master and be released. Basically, this is new v11 styles with the minimal adaptions from v10 components. Projects that use `carbon-components-svelte` can than use this release.

> _1.1 Bugfixing, feature parity_

Once v11 styles are released we might need to fix a few bugs, change a few more props – or we can start working towards [feature parity](https://github.com/carbon-design-system/carbon-components-svelte/discussions/1614).

> 2. Then, figure out an improved way for consumers to set up Carbon CSS in their projects

Here it is important to understand what problem we are solving. When discussing this, I think we should talk about if we want to support the v11 [`ClassPrefix`](https://react.carbondesignsystem.com/?path=/story/components-classprefix--default) component. This is related. Can library consumers choose their own class prefixes? How does that impact bundling of styles?

> 3. Ship these components, then continue work towards `carbon-components-svelte@v1`

At this point components should already be shipped. Improvements to developer experience and work towards feature parity can be parallel as long as we manage to make small increments that are mergeable to master.

> 4. Afterwards (out of scope for now), we set up `svelte-package`, types, and other v1 goals

As with point 3 I think typescript improvements can gradually be merged into master.

> If so, I'm on board with this.

I do think we are on the same page here.

File: src/DatePicker/DatePickerInput.svelte [hunk 1]
```diff
@@ -117,7 +117,7 @@
       class:bx--date-picker__input="{true}"
       class:bx--date-picker__input--invalid="{invalid}"
       class:bx--date-picker__input--sm="{size === 'sm'}"
-      class:bx--date-picker__input--xl="{size === 'xl'}"
+      class:bx--date-picker__input--lg="{size === 'lg' || size === 'xl'}"
```
Comment (gregorw) [inline src/DatePicker/DatePickerInput.svelte:120]: @theetrain I wonder if we should keep supporting legacy size `xl` for a while. What do you think?

**Comment by v3ss0n** [comment]
UIShell is broken withou this.

**Comment by theetrain** [comment]
Superseded by https://github.com/carbon-design-system/carbon-components-svelte/pull/1881

**Pull request closed without merging**